src/pdu.c

```diff
@@ -848,7 +853,8 @@ coap_pdu_parse_header(coap_pdu_t *pdu, coap_proto_t proto) {
 
 static int
 coap_pdu_parse_opt_csm(coap_pdu_t *pdu, uint16_t len) {
-  switch (pdu->code) {
+  switch ((uint8_t)pdu->code) {
+  /* (uint8_t) is to stop complaints about missing enums */
```

Comment:
Hm, `coap_pdu_code_t` has values larger than 255, so it will end up being at least an `int`. The compiler ought to warn here as well. I wonder if we could resolve this in a clean way by moving the COAP_SIGNALING_* codes into an `enum coap_pdu_signaling_code_t` and then do something like this, listing all values from `coap_pdu_signaling_code_t`:

```c
if (COAP_RESPONSE_CLASS(pdu->code) == 7) {
  coap_pdu_signaling_proto_t code = (coap_pdu_signaling_proto_t)pdu->code;
switch (code) {
   default:
     ...internal error, should not happen...
     ;
   case COAP_SIGNALING_CSM:
   ...
}
```

Reply:
Yes, I like the enum approach.  Will work on that

Reply:
Enum Change added and code pushed